### PR TITLE
Tidy up long-running and duplicated tests

### PR DESCRIPTION
**Pull request by tjof2**
### Description of the change
- `test_cluster.py` had far too many parameter combinations, including needless duplications, to the extent that it currently constitutes ~15% of all the unit tests by count. This has been reduced considerably for now, it can probably go further.
- `test_hologram_image.py` had unnecessary parameter combinations of `parallel=True/False` which have been removed.
- `test_tools.py` was >1500 lines long and really hard to parse. It has been split up into a number of new files to make it easier to improve the coverage etc. in future. The file names are open for renaming, I chose them based on what seemed like a natural grouping.

This doesn't need to go in v1.6 @francisco-dlp  if you don't want it to! 

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

**Comment by ericpre** [comment]
I was thinking exactly the same: `test_tools.py` has too many lines which is annoying and `test_cluster.py` takes too long!

**Comment by ericpre** [comment]
Most likely better to leave it after the release and then change the base branch to be merge in Rnp (after Rnm is merged into Rnp).

**Comment by tjof2** [comment]
Good idea @ericpre - looks like its shaved ~2mins off the test suite runtime according to Travis

**Comment by tjof2** [comment]
Switched to Rnp - codecov can be ignored here (v.small insignificant change)
